flanksource/flanksource-ui
# feat: add duration & status filters for check status history table

Closes #1015

### Test Plan
- login with your credentials
- go to health page by clicking on the health menu item on the sidebar
- you will be shown list of health checks, click on any one of them
- check details modal will be shown and observe new filters on status & duration are added for check status history table.
- modify the filters and observe filtered result set is as per the applied filters.

### Feature changes screenshot
<img width="1680" alt="Screenshot 2023-04-20 at 2 40 28 PM" src="https://user-images.githubusercontent.com/7310975/233318325-4c43c6cc-585a-40b1-a01e-2692fefe3258.png">

### Scope change details
* This [review comment](https://github.com/flanksource/flanksource-ui/pull/1090#issuecomment-1516855773) introduced scope change and introduced below changes.
  *  I have worked on the react select wrapper component to add support for allowing user input along with predefined options.
  * I have extracted a new `StatusHistoryFilters` component to keep the code clean.
  * added validations as we are allowing user inputs
  * refactored `StatusHistory` component a bit to integrate above changes.
